coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
+        return self._generate_linebreaks(file)
+
+    @property
     def unified_diff(self):
         """
         Generates a unified diff corresponding to this patch.
 
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
-                self.original,
-                self.modified,
+        list_unified_diff = list(difflib.unified_diff(
+                self._file,
+                self._modified(),
```

Comment:
why not using `self.original` and `self.modified`? Then you don't have to do `_generate_linebreaks`^^